skyportal/handlers/api/source.py

```diff
@@ -1062,8 +1083,40 @@ class Validator(Schema):
         if not save_summary:
             # Records are Objs, not Sources
             obj_list = []
-            for obj in query_results["sources"]:
+
+            # Load in all last_detected_at values at once
+            last_detected_at = Obj.last_detected_at(self.current_user)
+            query_results["sources"] = query_results["sources"].add_columns(
+                last_detected_at
+            )
+
+            # Load in all last_detected_mag values at once
+            last_detected_mag = Obj.last_detected_mag(self.current_user)
+            query_results["sources"] = query_results["sources"].add_columns(
+                last_detected_mag
+            )
+
+            # Load in all peak_detected_at values at once
+            peak_detected_at = Obj.peak_detected_at(self.current_user)
+            query_results["sources"] = query_results["sources"].add_columns(
+                peak_detected_at
+            )
+
+            # Load in all peak_detected_mag values at once
+            peak_detected_mag = Obj.peak_detected_mag(self.current_user)
+            query_results["sources"] = query_results["sources"].add_columns(
+                peak_detected_mag
+            )
```

Comment:
Instead of querying the last and peak detection info separately for each object in the result, we now do them in bulk by joining against the SQL expression versions of the properties instead. This speeds up the loop below dramatically.

Reply:
A really nice idea!